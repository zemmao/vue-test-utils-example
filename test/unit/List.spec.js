import { shallowMount } from '@vue/test-utils';
import List from '@/components/List';

describe(List.vue, () => {
  it('renders li for each item in props.items', () => {
    const items = ['1', '2'];
    const wrapper = shallowMount(List, { propsData: { items } });
    expect(wrapper.findAll('li')).toHaveLength(items.length);
  });

  it('matches snapshot', () => {
    const items = ['item1', 'item2'];
    const wrapper = shallowMount(List, { propsData: { items } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
