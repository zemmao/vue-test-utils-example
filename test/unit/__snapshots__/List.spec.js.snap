// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ul>
  <li>
    item1
  </li>
  <li>
    item2
  </li>
</ul>
`;
